backend/mechanismbackend/api/views.py

```diff
@@ -52,3 +57,64 @@ def create(self, request, *args, **kwargs):
             content = {'error': str(e)}
             return Response(content, status=status.HTTP_400_BAD_REQUEST)
 
+
+# dirty hack, try to find better way to achieve this
+class MechanismMatrix(APIView):
+    def get(self, request, format=None):
+        predicate = self.get_query_predicate()
+
+        matrix = []
+        matrix_inner = []
+
+        with connection.cursor() as cursor:
+            for input in ['R', 'T']:
+                for i in range(1, 4):
+                    i = str(i)
+                    for output in ['R', 'T']:
+                            for o in range(1, 4):
+                                o = str(o)
+
+                                cursor.execute('select input' + input + i + ', output' + output + o + ', Count(*) from api_mechanism ' +
+                                               predicate + ' group by input' + input + i + ', output' + output + o)
+                                rows = cursor.fetchall()
+
+                                num_mechanisms = 0
+                                for tpl in rows:
+                                    tpl_in, tpl_out, count = tpl
+                                    if tpl_in and tpl_out:
+                                        num_mechanisms = count
+                                        break
+
+                                matrix_inner.append(num_mechanisms)
+                                if output + o == 'T3':
+                                    matrix.append(matrix_inner)
+                                    matrix_inner = []
+        return Response(matrix)
+
+    def get_query_predicate(self):
+        params = list()
+        params.append(('inputR1', self.request.query_params.get('inputR1', None)))
+        params.append(('inputR2', self.request.query_params.get('inputR2', None)))
+        params.append(('inputR3',  self.request.query_params.get('inputR3', None)))
+        params.append(('inputT1',  self.request.query_params.get('inputT1', None)))
+        params.append(('inputT2',  self.request.query_params.get('inputT2', None)))
+        params.append(('inputT3', self.request.query_params.get('inputT3', None)))
+        params.append(('outputR1', self.request.query_params.get('outputR1', None)))
+        params.append(('outputR2', self.request.query_params.get('outputR2', None)))
+        params.append(('outputR3', self.request.query_params.get('outputR3', None)))
+        params.append(('outputT1', self.request.query_params.get('outputT1', None)))
+        params.append(('outputT2', self.request.query_params.get('outputT2', None)))
+        params.append(('outputT3', self.request.query_params.get('outputT3', None)))
+        params.append(('complete', self.request.query_params.get('complete', None)))
+        params.append(('transmission', self.request.query_params.get('transmission', None)))
+        params.append(('name', self.request.query_params.get('name', None)))
```

Comment:
Is it possible to use the same substring matching here that we use for the regular queries?